### ISSUE TITLE
upgrade Solid project to .Net 8

### ISSUE THREAD

**Issue by megahirt** [sillsdev/solid#6]
This issue is about seeing what it would take to upgrade the Solid program to net8-windows with the new windows forms.  I have no idea what that will take.  Migrating a relatively contained application like Solid to the new windows forms would be a valuable exercise that would inform whether or not we should attempt this for other SIL software.

**Comment by josephmyers** [comment]
@ethnos-men, these repo's are all managed by SIL and, in practice, us. Two of the repo's you've listed could be consumed by a .NET8 (windows) app, like Solid: SIL.DictionaryServices (.netstandard2.0) and L10NSharp (.net8.0-windows).

We are in the process of slowly upgrading the other libraries, mostly contained in [Libpalaso](https://github.com/sillsdev/libpalaso), but the jump from Framework 4.6.1 to .NET8 is pretty hefty, not to mention meeting some resistance due to its dropping support for some older OS's, etc. You can see more information [here](https://github.com/sillsdev/libpalaso/issues/1320), including our implementation strategy.

**Comment by josephmyers** [comment]
Kind of, yes. It gets tricky, though, since some of the libraries in libpalaso are Windows-specific, so they somewhat fall into the Application roadmap. Most of these are `SIL.Windows.Forms.*` libraries, with a couple exceptions, and they will need to target .NET8-Windows. This creates complication in our targeting, and our planned solution is to add in .NET8 on its own branch, eventually swapping it into master and keeping Framework on a branch. The idea is to make targeting .NET8 progressively easier, and Framework progressively harder.

Nuget publishing is done per project/library, yes, and it is able to manage our 3 different targeting platforms with a single assembly per version. At the WinForms level, whichever platform is on its own branch will have its own Nuget version (with the branch name integrated into the version name), though they can all share the same assembly name.

All libraries, cross-platform and WinForms, currently support Framework 4.8, so we're very close to completing Phase 1. However, there has been some recent discussion on if the branching plan is the best way to go about this. So we're still working this out.

I've updated the table names to be a tad bit clearer.

**Comment by josephmyers** [comment]
At the Hawaii conference, the team suggested it would in fact be useful to upgrade the SIL.Windows.Forms.* libraries to .NET8-Windows, with Solid as a good example/experiment of what this might look like. So I'm going to move forward with this.

**Comment by josephmyers** [comment]
Here is the PR for SIL.Windows.Forms:
https://github.com/sillsdev/libpalaso/pull/1358

I'm working on the remaining two now.

**Comment by josephmyers** [comment]
Yeah, we're getting really close to having something ready to test. But finding people who know it well enough to test it isn't straightforward either!

**Comment by megahirt** [comment]
@josephmyers I'd be willing to publish an alpha with a building .Net 8 Solid that passes cursory developer testing with some sample files.  The user base of Solid is really small, and we can put it in front of several users we know and ask them to give it a try.

**Comment by josephmyers** [comment]
Yeah, Chris, your installer GHA work is coming in handy. The latest installer is available to download [here](https://github.com/sillsdev/solid/actions/runs/13196544543). I performed cursory testing and have fixed all resultant issues. Maybe our userbase can find more. If you'd like me to send it to anyone, just let me know
